src/main/scala/services/consumers/StagedBatch.scala

```diff
@@ -41,9 +42,11 @@ trait StagedBatch:
 
   /**
    * Query that should be used to dispose of this batch data.
+   *
    * @return SQL query text
    */
-  def disposeExpr: String = s"DROP TABLE $name"
+  def disposeExpr(stagingTableName: String): String = s"DELETE FROM $stagingTableName WHERE true"
```

Comment:
```suggestion
  def disposeExpr(stagingTableName: String): String = s"DELETE FROM $stagingTableName WHERE true"
```
do `truncate table schema.name` instead